lib/roo/formatters/csv.rb

```diff
@@ -1,7 +1,7 @@
 module Roo
   module Formatters
     module CSV
-      def to_csv(filename = nil, separator = ",", sheet = default_sheet)
+      def to_csv(filename = nil, sheet = default_sheet, separator: ",")
```

Comment:
Coverage does randomly fluctuate a bit. I think we should do something like this:
```
def to_csv(filename = nil, old_separator = nil, old_sheet = nil, separator: ",", sheet: default_sheet)  
  if old_separator
    warn("[DEPRECATION] optional argument for separator is deprecated. Please use keyword argument :separator instead")
    separator= old_separator 
  end

  if old_sheet
    warn("[DEPRECATION] optional argument for sheet is deprecated. Please use keyword argument :sheet instead")
    sheet= old_sheet 
  end

  # ......
end
```

The above implemenation doesn't break existing code and we can safely depreceate multiple optional argument in favour of keyword argument.

Reply:
sorry but, why add 2 arguments that are not used at all, what props?

Reply:
Because we want to avoid making breaking changes. Doing so we support both old and new syntax. After this old syntax will still work but issue depreciation warning.

Deprecation warning informs what syntax and/or methods roo is planning to remove in future(probably in the next major version 3.0.0) which will break existing code. The suggested deprecation warning informs both the syntax change and suggests new syntax. 

Most of libraries in ruby ecosystem do not introduce breaking changes in minor or patch version and we also want to follow this as much as possible.

Reply:
now i see, thank you for detailed answer. done